### PR TITLE
Add support for binding to the specified CPU core

### PR DESCRIPTION
**Pull request by fouzhe**
AFL is now popular in academic research. To evaluate the effectiveness of improvements, we need to run two fuzzers (the former one and the improved one) for comparison.  To establish statistical significance of results, two fuzzers should run in similar environments. It would be better to select two CPU cores with similar initial conditions in the experiments.  However, the default CPU selection cannot support this.

### CODE REVIEW

File: afl-fuzz.c [hunk 1]
```diff
@@ -487,8 +490,20 @@ static void bind_to_free_cpu(void) {
   }
 
   closedir(d);
+  if (cpu_to_bind_given) {
+
+    if (0 <= cpu_to_bind && cpu_to_bind < cpu_core_count) {
```
Comment (Dor1s) [inline afl-fuzz.c:495]: `0 <= cpu_to_bind && ` can be removed since `cpu_to_bind` is unsigned

Comment (Dor1s) [reply afl-fuzz.c:495]: also, let's decrease the nested indentation by rewriting this in the following way:

```
    if (cpu_to_bind >= cpu_core_count)
      FATAL("The CPU core id to bind should be between 0 and %u", cpu_core_count - 1);

    if (cpu_used[cpu_to_bind])
      FATAL("The CPU core #%u to bind is not free!", cpu_to_bind);

    i = cpu_to_bind;
```

**Comment by fouzhe** [comment]
@Dor1s Thanks for your comments. I've updated with new patches for these points, please review again.

Thanks for your time!

File: afl-fuzz.c [hunk 2]
```diff
@@ -117,6 +119,7 @@ EXP_ST u8  skip_deterministic,        /* Skip deterministic stages?       */
            kill_signal,               /* Signal that killed the child     */
            resuming_fuzz,             /* Resuming an older fuzzing job?   */
            timeout_given,             /* Specific timeout given?          */
+           cpu_to_bind_given,         /* Specified cpu_to_bin given?      */
```
[the remark this answers is not in the document]
Comment (fouzhe) [reply afl-fuzz.c]: Done, thanks!

File: afl-fuzz.c (hunk 1 above)
Comment (fouzhe) [reply afl-fuzz.c]: Done, thanks!

**Pull request merged**